pkg/codegen/docs/gen.go

```diff
@@ -2016,33 +2017,38 @@ func (dctx *docGenContext) getMod(
 	add bool,
 ) *modContext {
 	modName := pkg.TokenToModule(token)
-	mod, ok := modules[modName]
+	return dctx.getModByModName(pkg, tokens.ModuleName(modName), tokenPkg, modules, tool, add)
+}
+
+func (dctx *docGenContext) getModByModName(
+	pkg schema.PackageReference,
+	modName tokens.ModuleName,
+	tokenPkg schema.PackageReference,
+	modules map[string]*modContext,
+	tool string,
+	add bool,
+) *modContext {
+	mod, ok := modules[string(modName)]
 	if !ok {
 		mod = &modContext{
 			pkg:           pkg,
-			mod:           modName,
+			mod:           string(modName),
 			tool:          tool,
 			docGenContext: dctx,
 		}
 
 		if modName != "" && codegen.PkgEquals(tokenPkg, pkg) {
-			parentName := path.Dir(modName)
-			// If the parent name is blank, it means this is the package-level.
-			if parentName == "." || parentName == "" {
```

Comment:
I guess the old code normalized "." and "" to "index", this is a little interesting, I think it might be an artifact of path.Dir and not actual pulumi module names? But we can preserve this oddity and normalize them to "index".

Reply:
I've seen tokens like "pkg:typ" which should normalise to "pkg:index:typ". The topmost module is either "" or "index".

Reply:
That's a great observation, let me fix it to handle both of those. I'm guessing "." is never a topmost module but was an artifact of using filepath.Dir

Reply:
Yes I believe so